broadcast.go

```diff
@@ -143,9 +143,6 @@ func (bc *broadcast) Rooms(connection Conn) []string {
 
 // AllRooms gives list of all rooms available for broadcast
 func (bc *broadcast) AllRooms() []string {
-	bc.lock.RLock()
```

Comment:
We need usage lock in method `getRoomsByConn(...)` in current package, because `bc.rooms` is critical section. Do you any time to fix it in this PR?

Reply:
> We need usage lock in method `getRoomsByConn(...)` in current package, because `bc.rooms` is critical section. Do you any time to fix it in this PR?

I don't get the point, both methods getRoomsByConn(...) and AllRooms() are only used by method bc.Rooms(), that its locks in the whole function.

Reply:
`All Rooms()` is public interface we won't remove lock/unlock sections for this rows. We will need remove lock/unlock operations at `Rooms()` method and add lock/unlock for private method - `get Rooms By Conn(...)` 

Upper's change is could be correctly for concurrency code